tests/MongoDB.Driver.Examples/CausalConsistencyExamples.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Examples
+{
+    public class CausalConsistencyExamples
+    {
+        [Fact]
+        public void Causal_Consistency_Example_1()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            var client = GetClient();
+
+            string itemsCollectionName = "items";
+            var testDatabase = client.GetDatabase("test");
+            testDatabase.DropCollection(itemsCollectionName);
+
+            // Start Causal Consistency Example 1
+            using (var session1 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+            {
+                var currenDate = DateTime.UtcNow.Date;
+                var items = client.GetDatabase("test", new MongoDatabaseSettings
+                    {
+                        ReadConcern = ReadConcern.Majority,
+                        WriteConcern = new WriteConcern(
+                            WriteConcern.WMode.Majority,
+                            TimeSpan.FromMilliseconds(1000))
+                    })
+                    .GetCollection<BsonDocument>("items");
+
+                items.UpdateOne(session1,
+                    Builders<BsonDocument>.Filter.And(new BsonDocument("sku", 111), new BsonDocument("end", BsonNull.Value)),
```

Comment:
This is an odd mix of using the Builder and plain BsonDocuments. Perhaps better is:

    Builders<BsonDocument>.Filter.And(
        Builders<BsonDocument>.Filter.Eq("sku", "111"),
        Builders<BsonDocument>.Filter.Eq("end", BsonNull.Value)),